admin_guide/managing_networking.adoc

```diff
@@ -144,6 +144,40 @@ $ systemctl restart atomic-openshift-master-api atomic-openshift-master-controll
 
 ----
 
+[[admin-guide-managing-networking-ingress]]
+== Setting up the HAProxy router to accept ingress objects
+
+By default, the HAProxy router only accepts route objects. You can set up the
+HAProxy router as a Kubernetes
+link:https://kubernetes.io/docs/concepts/services-networking/ingress/[Ingress
+Controller] so that it accepts both route and ingress objects. If configured,
+the same rules apply for claiming a domain for a namespace based on the creation
+time of the object. For example, the oldest route winning against any other
+claimants wanting to use the same namespace.
+
+For example, ownership of routes is based on the claims made for that host name with the same namespace. The oldest route wins against any other claims. 
+
+While this makes the router compatible with Kubernetes ingress
+objects, some caveats exist:
+
+* Ingress objects store the keys and certificates in secrets, so the router needs permission to read all secrets in the system.
+* Ingress objects only support edge termination for *https* routes.
+
+To configure an existing router to have Ingress support (assuming the default
+name of `router` for the deployment configuration and the service-account):
+
+. Set the `ROUTER_ENABLE_INGRESS` environment variable to `true`:
++
+----
+$ oc env dc router ROUTER_ENABLE_INGRESS=true`
+----
+
+. Add the cluster-admin role to the router:
++
+----
+$ oc adm policy add-role-to-user cluster-admin router`
```

Comment:
Below two role should be added to sa router, `oc adm policy add-role-to-user cluster-admin router` is invalid.
```
oc adm policy add-cluster-role-to-user cluster-reader system:serviceaccount:default:router
oc adm policy add-cluster-role-to-user system:openshift:controller:service-serving-cert-controller system:serviceaccount:default:router
```

Reply:
maybe `oc adm policy add-cluster-role-to-user cluster-reader -z router` also work here.